notary.go

```diff
@@ -5,3 +5,8 @@ package notary
 // confirmation), createNew will be true. Attempts is passed in so that implementers
 // decide how many chances to give to a human, for example.
 type PassRetriever func(keyName, alias string, createNew bool, attempts int) (passphrase string, giveup bool, err error)
+
+// CtxKey is a wrapper type for use in context.WithValue() to satisy golint
+// https://github.com/golang/go/issues/17293
+// https://github.com/golang/lint/pull/245
+type CtxKey string
```

Comment:
Non-blocking:  wondering if we can make constants for the 4 CTX keys we use:  `cryptoService`, `metaStore`, `repo` (maybe we should change this to "GUN" though), and `keyAlgorithm`.  

Even better is if we can unexport the type and just export the constants, but lint might complain.

At any rate, happy to have this done in another PR.